processor.go

```diff
@@ -311,20 +313,20 @@ func (p *processor) marshal(wr *prompb.WriteRequest) (bufOut []byte, err error)
 	return snappy.Encode(nil, b), nil
 }
 
-func (p *processor) dispatch(clientIP net.Addr, reqID uuid.UUID, m map[string]*prompb.WriteRequest) (res []result) {
+func (p *processor) dispatch(clientIP net.Addr, reqID uuid.UUID, tenantPrefix string, m map[string]*prompb.WriteRequest) (res []result) {
 	var wg sync.WaitGroup
 	res = make([]result, len(m))
 
 	i := 0
 	for tenant, wrReq := range m {
 		wg.Add(1)
 
-		go func(idx int, tenant string, wrReq *prompb.WriteRequest) {
+		go func(idx int, tenantPrefix string, tenant string, wrReq *prompb.WriteRequest) {
 			defer wg.Done()
 
-			r := p.send(clientIP, reqID, tenant, wrReq)
+			r := p.send(clientIP, reqID, tenantPrefix, tenant, wrReq)
 			res[idx] = r
-		}(i, tenant, wrReq)
+		}(i, tenantPrefix, tenant, wrReq)
```

Comment:
maybe just concat `tenant`+`tenantPrefix` here and not pass down to goroutine and `send()` as a parameter? if prefix is empty it won't make a difference